test/unit/helper-specs.js

```diff
@@ -322,4 +325,174 @@ describe('helpers', withMocks({fs}, function (mocks) {
+              spec resource 0x7f110004 io.appium.test:menu/conversation_multiuse: flags=0x00000000
+              spec resource 0x7f110005 io.appium.test:menu/toolbar_close_white: flags=0x00000000
+              spec resource 0x7f110006 io.appium.test:menu/toolbar_collection: flags=0x00000000
+              spec resource 0x7f110007 io.appium.test:menu/toolbar_sketch: flags=0x00000000
+              config (default):
+                resource 0x7f110000 io.appium.test:menu/conversation_header_menu_audio: t=0x03 d=0x000000b6 (s=0x0008 r=0x00)
+                  (string8) "res/menu/conversation_header_menu_audio.xml"
+                resource 0x7f110001 io.appium.test:menu/conversation_header_menu_collection: t=0x03 d=0x000000b7 (s=0x0008 r=0x00)
+                  (string8) "res/menu/conversation_header_menu_collection.xml"
+                resource 0x7f110002 io.appium.test:menu/conversation_header_menu_collection_searching: t=0x03 d=0x000000b8 (s=0x0008 r=0x00)
+                  (string8) "res/menu/conversation_header_menu_collection_searching.xml"
+                resource 0x7f110003 io.appium.test:menu/conversation_header_menu_video: t=0x03 d=0x000000b9 (s=0x0008 r=0x00)
+                  (string8) "res/menu/conversation_header_menu_video.xml"
+                resource 0x7f110004 io.appium.test:menu/conversation_multiuse: t=0x03 d=0x000000ba (s=0x0008 r=0x00)
+                  (string8) "res/menu/conversation_multiuse.xml"
+                resource 0x7f110005 io.appium.test:menu/toolbar_close_white: t=0x03 d=0x000000bb (s=0x0008 r=0x00)
+                  (string8) "res/menu/toolbar_close_white.xml"
+                resource 0x7f110006 io.appium.test:menu/toolbar_collection: t=0x03 d=0x000000bc (s=0x0008 r=0x00)
+                  (string8) "res/menu/toolbar_collection.xml"
+                resource 0x7f110007 io.appium.test:menu/toolbar_sketch: t=0x03 d=0x0000007f (s=0x0008 r=0x00)
+                  (string8) "res/menu/toolbar_sketch.xml"
+      `;
+      const aaptStrings = parseAaptStrings(aaptOutput, 'de-rDE');
+      aaptStrings.abc_action_bar_home_description.should.eql('Navigate "home"');
+      aaptStrings.calling__conversation_full__message.should.eql([
+        'Calls work in conversations with up to 1 person.',
+        'Calls work in conversations with up to %1$d people. "blabla"',
+      ]);
+    });
+  });
+
+  describe('parseAapt2Strings', function () {
+    it('should parse strings received from aapt2 output', function () {
+      const aapt2Output = `
+      Binary APK
+      Package name=io.appium.android.apis id=7f
+        type string id=0c entryCount=944
+          resource 0x7f0c0215 string/linear_layout_8_vertical
+            () "Vertical"
+            (fr) "Vertical"
+          resource 0x7f0c0216 string/linear_layout_8_horizontal
+            () "Horizontal"
+            (fr) "Horizontal"
+          resource 0x7f0c0217 string/linear_layout_8_top
```

Comment:
build tools 27 has a different format.
Haven't checked 26 etc, tho. 

```
kazu$ /Users/kazu/Library/Android/sdk/build-tools/27.0.3/aapt2 dump api.apk | grep "resource"
    spec resource 0x7f0e0087 io.appium.android.apis:string/admin_receiver_status_pw_changed
    spec resource 0x7f0e0088 io.appium.android.apis:string/admin_receiver_status_pw_failed
    spec resource 0x7f0e0089 io.appium.android.apis:string/admin_receiver_status_pw_succeeded
    spec resource 0x7f0e008a io.appium.android.apis:string/advanced_preferences
    spec resource 0x7f0e008b io.appium.android.apis:string/again
    spec resource 0x7f0e008c io.appium.android.apis:string/alarm_controller
    spec resource 0x7f0e008d io.appium.android.apis:string/alarm_service
    spec resource 0x7f0e008e io.appium.android.apis:string/alarm_service_finished
    spec resource 0x7f0e008f io.appium.android.apis:string/alarm_service_label
```

Reply:
So, https://github.com/appium/appium-adb/pull/467/files#diff-27a1c86133de4310373decfdad200a91R599 should be something like:

```diff
- if (trimmedLine.startsWith('resource')) {
+ if (/resource [\w]+/.test(trimmedLine)) {
```

?

Reply:
I've just checked aapt2 from build tools 27 and got the same output:

```
Nick:27.0.3 elf$ aapt2 dump ~/Desktop/ApiDemos-debug2.apk | grep resource
    resource 0x7f040000 anim/animator
    resource 0x7f040001 anim/animator_set
    resource 0x7f040002 anim/color_animator
    resource 0x7f040003 anim/cycle_7
    resource 0x7f040004 anim/fade
    resource 0x7f040005 anim/hold
```

what is the output of `aapt2 version` on your system? Mine shows 

```
$ aapt2 version
Android Asset Packaging Tool (aapt) 2:19
```

Reply:
mmm.
Mine was below

```
$ /Users/kazu/Library/Android/sdk/build-tools/27.0.3/aapt2 version
Android Asset Packaging Tool (aapt) 2:19
```

Reply:
These guys had the same output as me:
https://blog.csdn.net/iEearth/article/details/84632797

Perhaps, there is some hidden env flag, that influences that? I need to check aapt2 sources

Reply:
The recent aapt2 source should produce the same output: https://android.googlesource.com/platform/frameworks/base.git/+/master/tools/aapt2/Debug.cpp

Reply:
![image](https://user-images.githubusercontent.com/5511591/62409821-fb695c80-b617-11e9-8fa3-bd37367ea765.png)

My env got them via AS environment.
From 28.0.0, I was able to get the same one.

```
    resource 0x7f0e012f string/dismiss
    resource 0x7f0e0130 string/display_options_custom_button
    resource 0x7f0e0131 string/display_options_menu_item
    resource 0x7f0e0132 string/dive
```

One way is raising an error to update user env at least 28.0.0 (Maybe)?

Reply:
we default to aapt, so should be fine for now

Reply:
Sounds reasonable